src/silicone/multiple_infillers/infill_all_required_emissions_for_openscm.py

```diff
@@ -0,0 +1,340 @@
+
+    variable_leaders: list[str]
+        The name of the variable(s) found in to_fill which should be used to determine
+        the values of the other variables. For most infillers (including the default)
+        this list must contain only one entry. E.g. ["Emissions|CO2"]
+
+    required_variables_list: list[str]
+        The list of variables to infill. Each will be done separately. The default
+        behaviour (None option) will result in this being filled with the complete list
+        of required emissions.
+
+    cruncher : :class:
+        The class of cruncher to use to compute the infilled values. Defaults to
+        DatabaseCruncherQuantileRollingWindows, which uses the median value of a rolling
+        window. See the cruncher documentation for more details.
+
+    output_timesteps : list[int or datetime]
+        List of times at which to return infilled values. Will interpolate values in
+        between known data, but will not extend beyond the range of data provided.
+
+    infilled_data_prefix : str
+        A string that should be prefixed on all the variable names of the results
+        returned. Used to distinguish returned values from those input.
+
+    to_fill_old_prefix : str
+        Any string already found at the beginning of the variables names of the input
+        `to_fill` dataframe. This will be removed before comparing the variable names
+        with `database`.
+
+    check_data_returned : bool
+        If true, we perform checks that all desired data has been returned. Potential
+        reasons for failing this include requesting results at times outside our input
+        time range, as well as code bugs.
+
+    Returns
+    -------
+    :obj:`pyam.IamDataFrame`
+        The infilled dataframe (including input data) at requested times. All variables
+        now begin with infilled_data_prefix instead of to_fill_old_prefix.
+    """
+    # Use default arguments for unfilled options:
+    if output_timesteps is None:
+        output_timesteps = [2015] + list(range(2020, 2101, 10))
+    if required_variables_list is None:
+        required_variables_list = [
+            "Emissions|BC",
+            "Emissions|PFC|CF4",
+            "Emissions|PFC|C2F6",
+            "Emissions|PFC|C6F14",
+            "Emissions|CH4",
+            "Emissions|CO2|AFOLU",
+            "Emissions|CO2|Energy and Industrial Processes",
+            "Emissions|CO",
+            "Emissions|HFC|HFC134a",
+            "Emissions|HFC|HFC143a",
+            "Emissions|HFC|HFC227ea",
+            "Emissions|HFC|HFC23",
+            "Emissions|HFC|HFC32",
+            "Emissions|HFC|HFC43-10",
+            "Emissions|HFC|HFC245ca",
+            "Emissions|HFC|HFC125",
+            "Emissions|N2O",
+            "Emissions|NH3",
+            "Emissions|NOx",
+            "Emissions|OC",
+            "Emissions|SF6",
+            "Emissions|Sulfur",
+            "Emissions|VOC",
+        ]
```

Comment:
note: for later, this should be configurable.

Reply:
These are - they default to these values if we don't put anything else in. But I will now work on getting the aggregate values by direct calculation.

Reply:
re installation, good idea (I still need to write the installation instructions).